File: 2843_Count_Symmetric_Integers/solution.c
int countSymmetricIntegers(int low, int high) {
    int count = 0;

    for (int i = low; i <= high; i++) {
        char num[12];
        sprintf(num, "%d", i);
        int n = strlen(num);

        if (n % 2 != 0) continue;

        int mid = n / 2;

        int s1 = 0;
        for (int j = 0; j < mid; j++) {
            s1 += num[j] - '0';
        }

        int s2 = 0;
        for (int j = mid; j < n; j++) {
            s2 += num[j] - '0';
        }

        if (s1 == s2) {
            count++;
        }
    }

    return count;
}

File: 2843_Count_Symmetric_Integers/solution.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        count = 0
        
        for i in range(low, high + 1):
            num = str(i)
            n = len(num)

            if n % 2 != 0:
                continue

            mid = n // 2
            
            s1 = sum(int(num[j]) for j in range(mid))
            s2 = sum(int(num[k]) for k in range(mid, n))

            if s1 == s2:
                count += 1

        return count